File: backend/src/main/java/com/capstone/backend/service/UserService.java
package com.capstone.backend.service;

import com.capstone.backend.model.Users;
import com.capstone.backend.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService {
    @Autowired
    UserRepository userRepository;

    // Get all users
    public List<Users> getAll() {
        return userRepository.findAll();
    }

    // Get user with userId
    public Users getUserById(int userId) {
        return userRepository.getUserById(userId);
    }

    // Get user with userName
    public Users getByUserName(String userName) {
        return userRepository.getByUserName(userName);
    }

    // Add new user
    public String addUser(Users use) {
        String response;

        if (userRepository.addUser(use))
            response = "Successfully added";
        else
            response = "Something went wrong. Not added, please try again";
        return response;
    }

    // Update existing user
    public String updateUser(int userId, Users use) {
        String response;

        if (userRepository.updateUser(userId, use))
            response = "Successfully updated";
        else
            response = "Something went wrong. Not updated, please try again";
        return response;
    }

    // Delete user by userId
    public String deleteUser(int userId) {
        String response;

        if (userRepository.deleteUser(userId))
            response = "Successfully deleted";
        else
            response = "Something went wrong. Not deleted, please try again";
        return response;
    }

    // Verifies user login
    public ResponseEntity<?> login(Users user) {
        Users userdata = userRepository.getByUserName (user.getUserName()); //VERIFY IS USER EXISTS
        if(userdata != null)
        {
            if(user.getUserPassword ().equals(userdata.getUserPassword ()))
            {
                return new ResponseEntity<Users> (userdata, HttpStatus.OK);
            }
            else
            {
                return new ResponseEntity<> ("Wrong Password", HttpStatus.BAD_REQUEST);
            }
        }
        else
        {
            return new ResponseEntity<>(  "User does not exist!", HttpStatus.BAD_REQUEST);
        }
    }

    // Register new user
    public boolean register(Users user) {
        return userRepository.register(user);
    }

}

File: frontend/src/services/UserService.js
import axios from "axios";

const USER_BASE_REST_API_URL = "http://localhost:8080/api/v1/users";

class UserService {

  // Get all users
  getAllUsers() {
    return axios.get(USER_BASE_REST_API_URL + "/getAll");
  }

  // Create a new user
  createUser(user) {
    return axios.post(USER_BASE_REST_API_URL + "/addUser", user); 
  }

  // Get user by userId
  getUserbyId(userId) {
    return axios.get(USER_BASE_REST_API_URL + "/getUserById/" + userId); 
  }

  // Update an existing user
  updateUser(userId, user) {
    return axios.put(USER_BASE_REST_API_URL + "/updateUser/" + userId, user); 
  }

  // Delete a user by userId
  deleteUser(userId) {
    return axios.delete(USER_BASE_REST_API_URL + "/deleteUser/" + userId); 
  }

  // User login verification
  loginUser(user) {
    return axios.post("http://localhost:8080/api/v1/login/", user); 
  }
}

export default new UserService();